source/fuzz/fuzzer_pass_donate_modules.cpp

```diff
@@ -280,36 +281,71 @@ void FuzzerPassDonateModules::HandleTypesAndValues(
         // we go ahead and add a remapped version of the type declared by the
         // donor.
+        uint32_t component_type_id = type_or_value.GetSingleWordInOperand(0);
+        if (!original_id_to_donated_id->count(component_type_id)) {
+          // We did not donate the component type of this array type, so we
+          // cannot donate the array type.
+          continue;
+        }
         new_result_id = GetFuzzerContext()->GetFreshId();
         ApplyTransformation(TransformationAddTypeArray(
-            new_result_id,
-            original_id_to_donated_id->at(
-                type_or_value.GetSingleWordInOperand(0)),
+            new_result_id, original_id_to_donated_id->at(component_type_id),
             original_id_to_donated_id->at(
                 type_or_value.GetSingleWordInOperand(1))));
       } break;
+      case SpvOpTypeRuntimeArray: {
+        // A runtime array is allowed as the final member of an SSBO.  During
+        // donation we turn runtime arrays into fixed-size arrays.  For dead
+        // code donations this is OK because the array is never indexed into at
+        // runtime, so it does not matter what its size is.  For live-safe code,
+        // all accesses are made in-bounds, so this is also OK.
+        //
+        // The special OpArrayLength instruction, which works on runtime arrays,
+        // is rewritten to yield the fixed length that is used for the array.
+
+        uint32_t component_type_id = type_or_value.GetSingleWordInOperand(0);
+        if (!original_id_to_donated_id->count(component_type_id)) {
+          // We did not donate the component type of this runtime array type, so
+          // we cannot donate it as a fixed-size array.
+          continue;
+        }
+        new_result_id = GetFuzzerContext()->GetFreshId();
+        ApplyTransformation(TransformationAddTypeArray(
+            new_result_id, original_id_to_donated_id->at(component_type_id),
+            FindOrCreate32BitIntegerConstant(
+                GetFuzzerContext()->GetRandomSizeForNewArray(), false)));
+      } break;
       case SpvOpTypeStruct: {
         // Similar to SpvOpTypeArray.
-        new_result_id = GetFuzzerContext()->GetFreshId();
         std::vector<uint32_t> member_type_ids;
-        type_or_value.ForEachInId(
-            [&member_type_ids,
-             &original_id_to_donated_id](const uint32_t* component_type_id) {
-              member_type_ids.push_back(
-                  original_id_to_donated_id->at(*component_type_id));
-            });
+        for (uint32_t i = 0; i < type_or_value.NumInOperands(); i++) {
+          auto component_type_id = type_or_value.GetSingleWordInOperand(i);
+          if (!original_id_to_donated_id->count(component_type_id)) {
+            // We did not donate every member type for this struct type, so we
+            // cannot donate the struct type.
+            continue;
+          }
```

Comment:
Are these changes missing from the description of the PR?

Reply:
I had thought these changes (or at least some of them) were necessary due to the way OpRuntimeArrays are handled, but actually they are not, so I have removed them from the PR.